scripts/update/jellyfin.sh

```diff
@@ -68,15 +62,79 @@ if [[ -f /install/.jellyfin.lock ]]; then
         #
-        # Add the jellyfin official repository and key to our installation so we can use apt-get to install it jellyfin and jellyfin-ffmepg.
-        curl -s https://repo.jellyfin.org/$DIST_ID/jellyfin_team.gpg.key | gpg --dearmor > /usr/share/keyrings/jellyfin-archive-keyring.gpg 2>> "${log}"
-        echo "deb [signed-by=/usr/share/keyrings/jellyfin-archive-keyring.gpg arch=$(dpkg --print-architecture)] https://repo.jellyfin.org/$DIST_ID $DIST_CODENAME main" > /etc/apt/sources.list.d/jellyfin.list
+        # Make sure universe is enabled so that ffmpeg can be satisfied.
+        sudo add-apt-repository universe
+
+        # Handle some known alternative base OS values with 1-to-1 mappings
+        # Use the result as the repository base OS
+        ARCHITECTURE="$(dpkg --print-architecture)"
+        BASE_OS="$(awk -F'=' '/^ID=/{ print $NF }' /etc/os-release)"
+        case "${BASE_OS}" in
+            raspbian)
+                # Raspbian uses the Debian repository
+                REPO_OS="debian"
+                ;;
+            linuxmint)
+                # Linux Mint can either be Debian- or Ubuntu-based, so pick the right one
+                if grep -q "DEBIAN_CODENAME=" /etc/os-release &> /dev/null; then
+                    VERSION="$(awk -F'=' '/^DEBIAN_CODENAME=/{ print $NF }' /etc/os-release)"
+                    REPO_OS="debian"
+                else
+                    VERSION="$(awk -F'=' '/^UBUNTU_CODENAME=/{ print $NF }' /etc/os-release)"
+                    REPO_OS="ubuntu"
+                fi
+                ;;
+            neon)
+                # Neon uses the Ubuntu repository
+                REPO_OS="ubuntu"
+                ;;
+            *)
+                REPO_OS="${BASE_OS}"
+                VERSION="$(awk -F'=' '/^VERSION_CODENAME=/{ print $NF }' /etc/os-release)"
+                ;;
+        esac
+
         #
-        # install jellyfin and jellyfin-ffmepg using apt functions.
+        # Check if old, outdated repository for jellyfin is installed
+        # If old repository is found, delete it.
+        if [[ -f /etc/apt/sources.list.d/jellyfin.list ]]; then
+            echo_progress_start "Found old-style '/etc/apt/sources.list.d/jellyfin.list' configuration; removing it."
+            rm -f /etc/apt/sources.list.d/jellyfin.list
+            rm -f /etc/apt/keyrings/jellyfin.gpg
+            echo_progress_done "Removed old repository."
+        fi
+
+        #
+        # Add Jellyfin signing key if not already present
+        if [[ ! -f /etc/apt/keyrings/jellyfin.gpg ]]; then
+            echo_progress_start "> Did not find signing key. Adding it."
+            curl -fsSL https://repo.jellyfin.org/jellyfin_team.gpg.key | gpg --dearmor --yes --output /etc/apt/keyrings/jellyfin.gpg
+            echo_progress_done "Jellyfin Signing Key Added"
+        fi
+
+        #
+        # Install the Deb822 format jellyfin.sources entry
+        echo_progress_start "Adding Jellyfin repository to apt"
```

Comment:
Replicated code should be functionized (sources/functions/jellyfin), however I wouldn't block merging based on this, but it would be best-practices for long-term ease of maintenace.